src/BloomBrowserUI/bookEdit/js/readerToolsModel.ts

```diff
@@ -1002,6 +1002,21 @@ class ReaderToolsModel {
       return libsynphony.selectGPCWordsFromCache(desiredGPCs, knownGPCs, restrictToKnownGPCs, allowUpperCase, syllableLengths, selectedGroups, partsOfSpeech);
   }
 
+  static selectWordsFromAllowedLists(stageNumber: number): string[] {
+
+    var stages: ReaderStage[] = model.getSynphony().getStages(stageNumber);
+
+    var words: string[] = [];
+    for (var i=0; i < stages.length; i++) {
+      if (stages[i].allowedWords)
+        words = words.concat(stages[i].allowedWords);
+    }
+
+    words.sort();
+
+    return _.uniq(_.compact(words), true);
```

Comment:
Maybe. I'm waiting to see if the default javascript sort order is acceptable. Since it's a case sensitive sort (upper case before lower case), I'm guessing someone won't like it.